backpack/core/derivatives/mseloss.py

```diff
@@ -101,10 +69,64 @@ def hessian_mat_prod(mat):
 
         return hessian_mat_prod
 
-    def check_input_dims(self, module):
+    def _checks(self, module):
+        self._check_input_dims(module)
+
+    def _make_distribution(self, subsampled_input):
+        """Make the sampling distribution for the NLL loss form of MSE.
+
+        The log probabiity of the Gaussian distribution is proportional to
+        ¹/₍₂𝜎²₎∑ᵢ₌₁ⁿ (xᵢ−𝜇)². Because MSE = ∑ᵢ₌₁ⁿ(Yᵢ−Ŷᵢ)², this is
+        equivalent for samples drawn from a Gaussian distribution with
+        mean of the subsampled input and variance √0.5.
+
+        Args:
+            subsampled_input: input after subsampling
+
+        Returns:
+            torch.distributions Normal distribution with mean of
+        the subsampled input and variance √0.5
+
+        """
+        return Normal(mean(subsampled_input), tensor(sqrt(0.5)))
+
+    def _check_input_dims(self, module):
         """Raises an exception if the shapes of the input are not supported."""
         if not len(module.input0.shape) == 2:
             raise ValueError("Only 2D inputs are currently supported for MSELoss.")
 
-    def hessian_is_psd(self):
+    def hessian_is_psd(self) -> bool:
+        """Return whether cross-entropy loss Hessian is positive semi-definite.
+
+        Returns:
+            True
+
+        """
         return True
+
+    @staticmethod
+    def _get_mean_normalization(input: Tensor) -> int:
+        return input.numel()
+
+    def compute_sampled_grads(self, subsampled_input, mc_samples):
+        """Custom method to overwrite gradient computation for MeanSquareError Loss.
+
+        Because MSE = ∑ᵢ₌₁ⁿ(Yᵢ−Ŷᵢ)², the gradient is 2∑ᵢ₋₁ⁿ(Yᵢ−Ŷᵢ). Therefore, one can
+        sample this from a Gaussian distribution with a mean of 0 and a variance of √2.
+
+        Args:
+            subsampled_input: input after subsampling
+            mc_samples: number of samples
+
+        Returns:
+            sampled gradient
+
+        """
+        samples = normal(
```

Comment:
I think it would be better to use `make_distribution` here to avoid duplication

Reply:
These are actually slightly different distributions--here the variance is sqrt(2), but in make_distributions it is sqrt(1 / 2). The mean is also a bit different (0 vs subsampled_input). Alternatively, we could use _make_distribution to sample, then shift/scale the output? That may be a bit less clean theoretically though.